src/components/PostsList.tsx

```diff
@@ -1,85 +1,67 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+interface PostListProps {
+  posts: Post[],
+  postIdSelected: (id: number) => void,
+}
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<PostListProps> = ({
+  posts,
+  postIdSelected,
+}) => {
+  const [selectedId, setSelectedId] = useState(0);
```

Comment:
It's better to initialize selectedId with null instead of 0, as it's possible to have a post with id 0.